deploy/gatekeeper.yaml

```diff
@@ -364,7 +382,7 @@ webhooks:
   name: validation.gatekeeper.sh
   namespaceSelector:
     matchExpressions:
-    - key: control-plane
+    - key: gatekeeper-ignore
```

Comment:
This could be a breaking change for folks who are currently using the “control-plane” label on the kube-system namespace as a way to bypass the gatekeeper webhook.

Reply:
This is true. I'd suggest we continue to use the `gatekeeper-ignore` label as `control-plane` doesn't suggest "hey GK is ignoring this namespace", so that may surprise people.

I think the larger issue is that currently people would be unable to apply gatekeeper-ignore to any NS other than the one running Gatekeeper. Is this a use case that we want to support? If so, how should the allowed-to-be-ignored namespaces be specified? Flag?

Reply:
We could also make the name of the label configurable.

Reply:
What is the conclusion of this thread? Are there changes we want to make?

Reply:
> how should the allowed-to-be-ignored namespaces be specified? Flag?

Are you suggesting an array of strings representing namespaces? If so, we should include `kube-system, gatekeeper-system` by default. Updating this flag would require a restart of gatekeeper pod.

> make the name of the label configurable.

Yes and we should include `control-plane` by default to be backward compatible.

Reply:
Have people been applying the `control-plane` label to namespaces other than `gatekeeper-system`? If not, then this should be a backwards-compatible change as reapplying the manifest to update would add the new label to `gatekeeper-system`.

For those who have added the label to other namespaces, the webhook is still configurable so they have a few options:

1. Add the `control-plane` label selector back into the webhook, retaining the old behavior
2. Use the proposed "restrict these labels" flag to add "control-plane" back in, and the "these are the special namespaces" flag to enable use of `control-plane` on those namespaces
3. Add the `gatekeeper-ignore` label to any namespace that currently has a `control-plane` label and using the "these are the special namespaces" flag to enable use of `gatekeeper-ignore` on those namespaces.

It's also important to note that if people have been relying on the ignore behavior of the `control-plane` label previously, than this PR is already breaking, as they would need to add the excused namespaces to the "special namespaces" flag.

I don't think leaving `control-plane` as part of the default webhook configuration is the solution, as then the security hole this PR is meant to patch remains open.

Because the benefit of this PR requires breaking people who may have already added the `control-plane` label anyway and anyone else should be untouched, I'd like to take the opportunity to create a more explicitly named label that let's people know its danger and impact. It will also be less generic, as `control-plane` was a label that kubebuilder had generated.

Reply:
WRT allowing `kube-system` to be ignored by default...

It's plausible, but the principle of least privilege suggests that those who want to grant kube-system the power to ignore GK should explicitly opt-in.

Reply:
I like the intent of this PR. I know there are users using the control-plane label today to exclude namespaces from being processed by the validating webhook because the control-plane namespace label selector has been part of the gatekeeper deployment yaml in the past. 

If the goal is to get users off the control-plane label, then how do we signal the need to update the special label flag and the special namespace flag, and update the special label on all impacted namespaces? The current implementation will deploy/update gatekeeper successfully, but users won’t know the impact of this release after sometime when create and update of resources in special namespaces get denied.

Reply:
Incorporated feedback from mtg. I should probably also update the README

Reply:
Okay, should be ready to review.